File: pkg/models/session.go
package models

import "time"

type Session struct {
	ID             string
	UserID         string
	AccessToken    string
	RefreshToken   string
	UserAgent      string
	IPAddress      string
	DeviceID       string
	LastActivityAt time.Time
	CreatedAt      time.Time
	ExpiresAt      time.Time
	DeactivatedAt  *time.Time
	Location       string
}

File: admin-ui/src/types/session.ts
// SessionResponse is not in swagger spec yet — defined manually to match Go SessionResponse
export interface SessionResponse {
  id: string;
  user_id: string;
  user_agent: string;
  ip_address: string;
  device_id: string | null;
  last_activity_at: string | null;
  created_at: string;
  expires_at: string;
  deactivated_at: string | null;
  location: string;
  status: "active" | "expired" | "deactivated";
}
